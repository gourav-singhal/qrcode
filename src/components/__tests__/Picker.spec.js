import React from 'react';
import TestRenderer from 'react-test-renderer';

import { Picker } from '../index';

describe('Picker Component', () => {
  // it('renders as expected on datetime', () => {
  //   const props = {
  //     type: 'datetime',
  //     key: 1,
  //     placeholder: 'hello',
  //     inputValue: 123,
  //   };

  //   const wrapper = TestRenderer.create(<Picker {...props} />);
  //   expect(wrapper.toJSON()).toMatchSnapshot();
  // });

  it('renders as expected on picker', () => {
    const props = {
      type: 'picker',
      key: 1,
      items: [{ id: 1, label: 'hello' }, { id: 2, label: 'man' }],
    };

    const wrapper = TestRenderer.create(<Picker {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
