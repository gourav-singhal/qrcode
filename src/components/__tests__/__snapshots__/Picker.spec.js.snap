// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Component renders as expected on picker 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      undefined,
      Object {},
      Object {},
      Object {},
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onFocus={[Function]}
      placeholder="Value"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "color": "#626262",
            "flexGrow": 1,
            "marginBottom": 15,
            "padding": 20,
            "shadowColor": "#2E374D",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          },
          undefined,
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      transparent={false}
      visible={false}
    >
      <View
        centerV={true}
        flex-1={true}
        paddingH-15={true}
        style={
          Array [
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            Object {
              "paddingHorizontal": 15,
            },
            Object {},
            Object {
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          h2={true}
          marginB-20={true}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "left",
              },
              Object {
                "fontFamily": "Muli-SemiBold",
                "fontSize": 18,
                "lineHeight": 22,
              },
              undefined,
              Object {
                "marginBottom": 20,
              },
              Object {
                "textAlign": "center",
              },
              undefined,
            ]
          }
        />
        <View>
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "hello",
                  "textColor": undefined,
                  "value": undefined,
                },
                Object {
                  "label": "man",
                  "textColor": undefined,
                  "value": undefined,
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 50,
              "opacity": 1,
              "shadowColor": "#FC6821",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4294947695,
                4294933308,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 1,
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 50,
                  "padding": 20,
                },
                undefined,
                Object {
                  "marginTop": 20,
                },
              ]
            }
          >
            <Text
              h2={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "left",
                  },
                  Object {
                    "fontFamily": "Muli-SemiBold",
                    "fontSize": 18,
                    "lineHeight": 22,
                  },
                  undefined,
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Save
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
